js/src/common/utils/arrayFlatPolyfill.ts

```diff
@@ -3,23 +3,25 @@
 // Needed to provide support for Safari on iOS < 12
 
 if (!Array.prototype['flat']) {
-  Array.prototype['flat'] = function flat(depth?: number) {
+  Array.prototype['flat'] = function flat(this: any, depth?: number) {
     const realDepth = isNaN(depth as any) ? 1 : depth;
 
     return realDepth
       ? Array.prototype.reduce.call(
           this,
           function (acc, cur) {
             if (Array.isArray(cur)) {
-              acc.push.apply(acc, flat.call(cur, realDepth - 1));
+              (acc as Array<any>).push.apply(acc, flat.call(cur, realDepth - 1));
```

Comment:
Can this not be done in the `function (acc: )` argument type hinting?

Reply:
I would love to if I could figure out what Typescript was continuously complaining about...

I think `Array.prototype.reduce.call` expects the input to be anything, as it's a generic function, so it just uses `any` as the types for the accumulator and current value, which plays havoc when we want to use `.push`.

Reply:
Yeah I can't easily read that xD

Reply:
Y e a h...

I might just type the function args then do `as any` on the callback to prevent any complaints. That makes it a *bit* cleaner.